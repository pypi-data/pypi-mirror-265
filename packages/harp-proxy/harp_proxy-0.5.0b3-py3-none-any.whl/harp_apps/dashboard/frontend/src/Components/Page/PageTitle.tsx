import { ReactNode } from "react"

import { H1, P } from "ui/Components/Typography"

interface PageTitleProps {
  title?: string
  description?: string
  children?: ReactNode
}

export function PageTitle({ description, title, children }: PageTitleProps) {
  return (
    <>
      {title ? (
        <div className="mt-4 flex">
          <div className="flex flex-col">
            <H1>{title}</H1>
            {description ? <P>{description}</P> : null}
          </div>
          {children ? <>{children}</> : null}
        </div>
      ) : null}
    </>
  )
}
