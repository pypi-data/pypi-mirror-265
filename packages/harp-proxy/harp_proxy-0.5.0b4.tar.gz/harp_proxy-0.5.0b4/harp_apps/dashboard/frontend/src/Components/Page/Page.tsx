import * as Sentry from "@sentry/browser"
import { ReactNode } from "react"
import { ErrorBoundary, FallbackProps } from "react-error-boundary"

export function Error(props: FallbackProps) {
  return (
    <div>
      Woopsie!
      <div className="card my-5">
        <div className="card-header">
          <p>
            An error has occurred in this component.{" "}
            <span
              style={{ cursor: "pointer", color: "#0077FF" }}
              onClick={() => {
                window.location.reload()
              }}
            >
              Reload this page
            </span>{" "}
          </p>
        </div>

        <div className="card-body">
          <details className="error-details">
            <summary>Click for error details</summary>
            <pre>
              {/* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */}
              {props.error.stack}
            </pre>
          </details>
        </div>

        <button className="bg-primary text-light" onClick={() => Sentry.showReportDialog()}>
          Report feedback
        </button>
      </div>
      <br />
      <pre style={{ color: "red" }}>{(props.error as Error).message}</pre>
    </div>
  )
}

interface PageProps {
  children: ReactNode
  title?: ReactNode
}

export function Page({ children, title }: PageProps) {
  return (
    <ErrorBoundary FallbackComponent={Error}>
      <main>
        {title ?? null}
        <div className="mt-4">
          <div className="sm:-mx-6 lg:-mx-8">
            <section className="inline-block min-w-full max-w-full py-2 align-middle sm:px-6 lg:px-8">
              {children}
            </section>
          </div>
        </div>
      </main>
    </ErrorBoundary>
  )
}
