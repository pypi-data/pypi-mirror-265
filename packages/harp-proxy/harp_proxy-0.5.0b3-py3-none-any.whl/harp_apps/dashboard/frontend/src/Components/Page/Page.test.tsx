import { render, screen } from "@testing-library/react"
import { ErrorBoundary } from "react-error-boundary"
import { afterEach, describe, expect, test, vi } from "vitest"

import { Error as ErrorFunc, Page } from "./Page.tsx"
import { PageTitle } from "./PageTitle.tsx"

describe("Page", () => {
  afterEach(() => {
    vi.restoreAllMocks()
  })

  test("renders without error", () => {
    const { container } = render(
      <Page children={<div />} title={<PageTitle title="Test Title" description="Test Description" />} />,
    )
    expect(container).toMatchSnapshot()
  })

  test("renders title and description", () => {
    render(<Page children={<div />} title={<PageTitle title="Test Title" description="Test Description" />} />)
    expect(screen.getByText("Test Title")).toBeInTheDocument()
    expect(screen.getByText("Test Description")).toBeInTheDocument()
  })

  test("renders Error component when child throws error", () => {
    vi.spyOn(console, "error").mockImplementation(() => {})
    const Child = () => {
      throw new Error("Test error")
    }

    render(
      <ErrorBoundary FallbackComponent={ErrorFunc}>
        <Page children={<Child />} />
      </ErrorBoundary>,
    )

    expect(screen.getByText("Woopsie!")).toBeInTheDocument()
    expect(screen.getByText("An error has occurred in this component.")).toBeInTheDocument()
  })
})
