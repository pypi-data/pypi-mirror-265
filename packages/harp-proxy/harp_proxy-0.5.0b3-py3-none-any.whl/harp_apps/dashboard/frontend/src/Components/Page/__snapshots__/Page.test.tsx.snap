// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > renders without error 1`] = `
<div>
  <main>
    <div
      class="mt-4 flex"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="css-9lrcch"
        >
          Test Title
        </h1>
        <p
          class="css-11bp3eh"
        >
          Test Description
        </p>
      </div>
    </div>
    <div
      class="mt-4"
    >
      <div
        class="sm:-mx-6 lg:-mx-8"
      >
        <section
          class="inline-block min-w-full max-w-full py-2 align-middle sm:px-6 lg:px-8"
        >
          <div />
        </section>
      </div>
    </div>
  </main>
</div>
`;
